### ISSUE TITLE
CSS isolation ::deep not working with > selector

### ISSUE THREAD

**Issue by doyce** [dotnet/aspnetcore#25686]
I have a component containing the following code:

```
<div class="section">
    <ChildComponent />
    <ChildComponent />
    <ChildComponent />
</div>
```

with following css selector to apply styling to all child elements:

```
.section > * {
    // some styling
}
```

Now the issue is, that after CSS isolation was introduced, the selector is scoped using an attribute:

```
.section > *[b-xyz] {
    // some styling
}
```

but child components do not contain this attribute, so the selector is not applied to them. I understand that I need to use ::deep selector to bypass the scoping, supposingly like this:

```
.section::deep > * {
    // some styling
}

```
but the :deep selector is not replaced by attribute selector in this case. The output inside scoped.styles.css still contains ::deep.

**Comment by barahonajm** [comment]
I did not test it but I believe you are missing a space between section and ::deep:
```
.section ::deep > * {
    // some styling
}

```

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
This will be addressed in the upcoming 5.0 RC1 SDK. Also please make sure you apply @barahonajm 's suggestion above.

**Comment by doyce** [comment]
I tried also adding a space before but it doesn't make a difference. I'm looking forward to trying the RC1 SDK then :)
